cli/install.go

```diff
@@ -223,114 +301,84 @@ func (z *ZVM) InstallZls(version string) error {
 
 	arch, osType := zigStyleSysInfo()
+	expectedArchOs := fmt.Sprintf("%v-%v", arch, osType)
 
 	filename := "zls"
 	if osType == "windows" {
 		filename += ".exe"
 	}
 
 	// master does not need unzipping, zpm just serves full binary
-	var releaseUrl string
+	shouldUnzip := true
 	if version == "master" {
-		releaseUrl = "https://api.github.com/repos/zigtools/zls/releases/latest"
-	} else {
-		// build url for tagged version
-		releaseUrl = fmt.Sprintf("https://api.github.com/repos/zigtools/zls/releases/tags/%v", version)
+		shouldUnzip = false
 	}
 
-	// get release information
-	resp, err := http.Get(releaseUrl)
+	downloadUrl, err := getZLSDownloadUrl(version, expectedArchOs)
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
 
-	var releaseBuffer bytes.Buffer
-	_, err = releaseBuffer.ReadFrom(resp.Body)
+	request, err := http.NewRequest("GET", downloadUrl, nil)
 	if err != nil {
 		return err
 	}
 
-	// some github releases use tar.gz, some tar.xz
-	expectedArchOs := fmt.Sprintf("%v-%v", arch, osType)
-	zipName := ""
-	var taggedReleaseResponse githubTaggedReleaseResponse
-	// getting list of assets
-	if err := json.Unmarshal(releaseBuffer.Bytes(), &taggedReleaseResponse); err != nil {
-		return err
-	}
+	request.Header.Set("User-Agent", "zvm "+meta.VERSION)
 
-	// getting platform information
-	downloadUrl := ""
-	for _, asset := range taggedReleaseResponse.Assets {
-		if strings.Contains(asset.Name, expectedArchOs) {
-			downloadUrl = asset.Url
-			zipName = asset.Name
-			break
-		}
-	}
-
-	// couldn't find the file
-	if downloadUrl == "" {
-		return fmt.Errorf("could not find zls-%v", expectedArchOs)
-	}
-
-	if version == "master" {
-		downloadUrl = fmt.Sprintf("https://zig.pm/zls/downloads/%v/bin/zls", expectedArchOs)
-	}
-
-	client := &http.Client{}
-
-	// download tarball to temp
-	req, err := http.NewRequest("GET", downloadUrl, nil)
-	if err != nil {
-		return err
-	}
-	req.Header.Set("Accept", "application/octet-stream")
-	resp, err = client.Do(req)
+	response, err := http.DefaultClient.Do(request)
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
-
-	// creating file to place contents
-	tempName := filename
-	if version != "master" {
-		_, pathEnding, _ := strings.Cut(zipName, ".")
-		tempName = "*." + pathEnding
-	}
-
-	tempDir, err := os.CreateTemp(z.zvmBaseDir, tempName)
-	if err != nil {
-		return err
-	}
-
-	defer tempDir.Close()
-	defer os.RemoveAll(tempDir.Name())
+	defer response.Body.Close()
 
 	pbar := progressbar.DefaultBytes(
-		int64(resp.ContentLength),
+		int64(response.ContentLength),
 		"Downloading ZLS",
 	)
 
-	if _, err := io.Copy(io.MultiWriter(pbar, tempDir), resp.Body); err != nil {
-		return err
-	}
-	// untar to destination
-	fmt.Println("Extracting ZLS...")
 	versionPath := filepath.Join(z.zvmBaseDir, version)
-	if version == "master" {
-		if err := os.Rename(tempDir.Name(), filepath.Join(versionPath, filename)); err != nil {
+	binaryLocation := filepath.Join(versionPath, filename)
```

Comment:
This can just be reduced into one call when constructing `versionPath`. Join supports variadic arguments.